react/features/recording/components/LiveStream/LiveStreamButton.web.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+
+import { connect } from 'react-redux';
+
+import { translate } from '../../../base/i18n';
+import { getLocalParticipant } from '../../../base/participants';
+
+import AbstractLiveStreamButton, {
+    _mapStateToProps as _abstractMapStateToProps,
+    type Props as AbstractProps
+} from './AbstractLiveStreamButton';
+
+declare var interfaceConfig: Object;
+
+type Props = AbstractProps & {
+
+    /**
+     * True if the button should be disabled, false otherwise.
+     *
+     * NOTE: On web, if the feature is not disabled on purpose, then we still
+     * show the button but disabled and with a tooltip rendered on it,
+     * explaining why it's not available.
+     */
+    _disabled: boolean,
+
+    /**
+     * Tooltip for the button when it's disabled in a certain way.
+     */
+    _liveStreamDisabledTooltipKey: ?string
+}
+
+/**
+ * An implementation of a button for starting and stopping live streaming.
+ */
+class LiveStreamButton extends AbstractLiveStreamButton<Props> {
+    iconName = 'icon-public';
+    toggledIconName = 'icon-public';
+
+    /**
+     * Constructor of the component.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.tooltip = props._liveStreamDisabledTooltipKey;
```

Comment:
What is the reason for storing the tooltip on the instance instead of accessing props._liveStreamDisabledTooltipKey directly wherever it's being referenced?

Reply:
Good question. ```AbstractButton``` was implemented that way originally, I guess it was not a plan at that time that these text values come from props too many times.